patches/0004-Output-to-hashed-mappings.patch

```diff
@@ -0,0 +1,1622 @@
++					accessWidener,
++					intermediaryRemapper.getEnvironment().getRemapper(),
+ 					MappingsNamespace.INTERMEDIARY.toString(),
+ 					MappingsNamespace.NAMED.toString()
+ 			);
+-			AccessWidenerReader transitiveReader = new AccessWidenerReader(new TransitiveOnlyFilter(remappingVisitor));
++			AccessWidenerReader intermediaryTransitiveReader = new AccessWidenerReader(new TransitiveOnlyFilter(intermediaryRemappingVisitor));
++
++			final Map<String, AccessWidenerReader> awReaderMap = new HashMap<>();
++			awReaderMap.put(MappingsNamespace.HASHED.toString(), hashedTransitiveReader);
++			awReaderMap.put(MappingsNamespace.INTERMEDIARY.toString(), intermediaryTransitiveReader);
+ 
+ 			for (AccessWidenerFile accessWidenerFile : transitiveAccessWideners) {
+ 				project.getLogger().info("Reading transitive access widener from {}", accessWidenerFile.modId());
+-				transitiveReader.read(accessWidenerFile.content());
++				String awNamespace = AccessWidenerReader.readHeader(accessWidenerFile.content()).getNamespace();
++				awReaderMap.get(awNamespace).read(accessWidenerFile.content());
+ 			}
+ 		} finally {
+-			tinyRemapper.finish();
++			hashedRemapper.finish();
++			intermediaryRemapper.finish();
+ 		}
+ 
+ 		return accessWidener;
+ 	}
+ 
+-	private TinyRemapper createTinyRemapper() {
++	private TinyRemapper createTinyRemapper(MappingsNamespace intermediateMappings) {
+ 		try {
+-			TinyRemapper tinyRemapper = TinyRemapperHelper.getTinyRemapper(project, "intermediary", "named");
++			TinyRemapper tinyRemapper = TinyRemapperHelper.getTinyRemapper(project, "hashed", "named");
+ 
+ 			tinyRemapper.readClassPath(TinyRemapperHelper.getMinecraftDependencies(project));
+ 
+-			for (Path minecraftJar : extension.getMinecraftJars(MappingsNamespace.INTERMEDIARY)) {
++			for (Path minecraftJar : extension.getMinecraftJars(intermediateMappings)) {
+ 				tinyRemapper.readClassPath(minecraftJar);
+ 			}
+ 
+ 			return tinyRemapper;
+ 		} catch (IOException e) {
+-			throw new RuntimeException("Failed to create tiny remapper for intermediary->named", e);
++			throw new RuntimeException("Failed to create tiny remapper for hashed->named", e);
+ 		}
+ 	}
+ 
+diff --git a/src/main/java/net/fabricmc/loom/configuration/accesswidener/TransitiveAccessWidenerMappingsProcessor.java b/src/main/java/net/fabricmc/loom/configuration/accesswidener/TransitiveAccessWidenerMappingsProcessor.java
+index e82ea83fc7ee735cc1922c35dbc0801bd7709a74..24d21d1cb4344032225a44724d2662ea040f3709 100644
+--- a/src/main/java/net/fabricmc/loom/configuration/accesswidener/TransitiveAccessWidenerMappingsProcessor.java
++++ b/src/main/java/net/fabricmc/loom/configuration/accesswidener/TransitiveAccessWidenerMappingsProcessor.java
+@@ -48,8 +48,8 @@ public record TransitiveAccessWidenerMappingsProcessor(Project project) implemen
+ 			return false;
+ 		}
+ 
+-		if (!MappingsNamespace.INTERMEDIARY.toString().equals(mappings.getSrcNamespace())) {
```

Comment:
Doesn't this break depending on mods with transitive AWs in Intermediary?

Reply:
The mapping tree passed is just the normal loom mapping provider but with the source namespace switched to hashed, it's not the namespace of the transitive AW.

Reply:
Wait you might be right actually, I'll have to do a little more digging.